dist/common/scripts/scylla_util.py

```diff
@@ -575,18 +613,40 @@ def is_gentoo_variant():
 def redhat_version():
     return distro.version()
 
-def is_ec2():
-    if os.path.exists('/sys/hypervisor/uuid'):
-        with open('/sys/hypervisor/uuid') as f:
-            s = f.read()
-        return True if re.match(r'^ec2.*', s, flags=re.IGNORECASE) else False
-    elif os.path.exists('/sys/class/dmi/id/board_vendor'):
-        with open('/sys/class/dmi/id/board_vendor') as f:
-            s = f.read()
-        return True if re.match(r'^Amazon EC2$', s) else False
+
+def get_text_from_path(fpath):
+    board_vendor_path = Path(fpath)
+    if board_vendor_path.exists():
+        return board_vendor_path.read_text().strip()
+    return ""
+
+def match_patterns_in_files(list_of_patterns_files):
+    for pattern, fpath in list_of_patterns_files:
+        if re.match(pattern, get_text_from_path(fpath), flags=re.IGNORECASE):
+            return True
     return False
 
 
+def is_ec2():
+    patterns_files = ((r"^Amazon EC2$", "/sys/class/dmi/id/board_vendor"),
+                      (r"^ec2.*", "/sys/hypervisor/uuid"))
+    return match_patterns_in_files(patterns_files)
```

Comment:
This looks fragile. Is there no way to do this from the metadata server?

Reply:
this is what AWS recommends. You can read more here: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/identify_ec2_instances.html

Reply:
The second one is not accurate. There is a 1:4000 chance of a random uuid starting with "ec2". They even document it as "probably".

Reply:
I think it's better to use the instance metadata, since we have to read it anyway.

Reply:
fixed.